metrics_ci_integration/lib/jenkins/client/model/jenkins_query_limits.dart

```diff
@@ -0,0 +1,177 @@
+import 'package:ci_integration/jenkins/client/util/number_validator.dart';
+
+/// A class representing a range specifier for array-type properties in
+/// Jenkins API requests.
+class JenkinsQueryLimits {
+  /// The lower bound of a range specified.
+  final int lower;
+
+  /// The upper bound of a range specified.
+  final int upper;
+
+  /// Creates a range specifier with given [lower] and [upper] bounds.
+  const JenkinsQueryLimits._({this.lower, this.upper});
+
+  /// Checks [number] to be valid string representing a range-specifier edge.
+  static bool _checkValueString(
+    String number, {
+    bool allowEmpty = true,
+  }) {
+    if (number == null) return false;
+    if (allowEmpty && number.isEmpty) return true;
+    final result = int.tryParse(number);
+    return result != null && result >= 0;
+  }
+
+  /// Creates an empty range specifier with no edges.
+  ///
+  /// Requests with no range-specifier set will fetch the full data list.
+  const JenkinsQueryLimits.empty() : this._();
+
+  /// Parses a [query] with range-specifier to an instance of [JenkinsQueryLimits].
+  ///
+  /// Throws [FormatException] if parsing fails.
+  factory JenkinsQueryLimits.fromQuery(String query) {
```

Comment:
Should this function throw an ArgumentException if the upper bound is less then lower bound? I mean, if the `{3,1}` query passed, for example. Also, maybe we can add a test for this case?